lib/go-parser/parser.ts

```
@@ -0,0 +1,135 @@
+    modules,
+    packages,
+  );
+  return createDepGraph(name, packageVersionTable);
+}
+
+/**
+ * Package name consist of module name and path to package.
+ * This function matches package to modules and their versions
+ * @param moduleVersionTable
+ * @param packages
+ */
+function matchModuleToPackage(
+  moduleVersionTable: any,
+  packages: GoPackage[],
+): PackageVersionTable {
+  const resultTable: PackageVersionTable = {};
+  const packagesToParse: GoPackage[] = [];
+  for (const pack of packages) {
+    const moduleVersion = moduleVersionTable[pack];
+
+    if (!moduleVersion) {
+      packagesToParse.push(pack);
+      continue;
+    }
+
+    resultTable[pack] = moduleVersion;
+  }
+
+  for (const packToParse of packagesToParse) {
+    const pack = packToParse.substring(0, packToParse.lastIndexOf("/"));
+    const moduleVersion = moduleVersionTable[pack];
+
+    if (!moduleVersion) {
+      if (pack) {
+        packagesToParse.push(pack);
+      }
+
+      continue;
+    }
+
+    resultTable[pack] = moduleVersion;
+  }
+
+  return resultTable;
+}
+
+function createDepGraph(
+  name: string,
+  packageVersionTable: PackageVersionTable,
+): depGraph.DepGraph {
+  const goModulesDepGraph = new depGraph.DepGraphBuilder(
+    { name: DEP_GRAPH_TYPE },
+    { name },
+  );
+
+  for (const [name, version] of Object.entries(packageVersionTable)) {
+    const nodeId = `${name}@${version}`;
+    goModulesDepGraph.addPkgNode({ name, version }, nodeId);
+    goModulesDepGraph.connectDep(goModulesDepGraph.rootNodeId, nodeId);
+  }
+
+  return goModulesDepGraph.build();
+}
+
+/**
+ * Build depGraphs for each Go executable
+ * @param filePathToContent
+ */
+export async function goModulesToScannedProjects(
+  filePathToContent: FilePathToElfContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  for (const [filePath, goBinary] of Object.entries(filePathToContent)) {
```

Is it possible any of these steps in the loop could throw? Or we want to surface any errors/bugs in our logic instead of doing a best-effort attempt?

---

Can't find a place with a possible error. My only guess we might get thrown out if `address`, `offset`, etc would be out of range of `Number`, but don't think it would actually be there.